include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp

```diff
@@ -190,8 +190,6 @@ class two_dimensional_matrix
      */
     constexpr reference operator[](matrix_coordinate const & coordinate) noexcept
     {
-        assert(coordinate.col >= 0u);
-        assert(coordinate.row >= 0u);
```

Comment:
Hm but are we sure that this is always an unsigned type?

Reply:
https://github.com/seqan/seqan3/blob/f54cb5e6048eb2c205e0dc913fb44166c94357cf/include/seqan3/alignment/matrix/detail/matrix_coordinate.hpp#L135-L137

yes

Reply:
@rrahn looks fine from my side but maybe you take a second look because I don't know the reason why `matrix_index<size_t>` is templatized and it might be changed/used differently?

Reply:
I asked him when I removed it, and he said it's always unsigned and defined as size_t

Reply:
alright